include/memory_allocator.h

```diff
@@ -0,0 +1,27 @@
+#ifndef __MEMORY_ALLOCATOR_H__
+#define __MEMORY_ALLOCATOR_H__
+
+typedef struct malloc_pool malloc_pool_t;
+
+/* Defines a local pool of memory for use by a subsystem. */
+#define MALLOC_DEFINE(pool, desc)     \
+    malloc_pool_t pool[1] = {         \
+        {{NULL}, MAGIC, desc} \
+    };
+
+#define MALLOC_DECLARE(pool) \
+    extern malloc_pool_t pool[1]
+
+/* Flags to malloc */
+#define M_WAITOK    0x0000 /* ignore for now */
+#define M_NOWAIT    0x0001 /* ignore for now */
+#define M_ZERO      0x0002 /* clear allocated block */
+
+void malloc_add_arena(malloc_pool_t *mp, void *start, size_t size);
```

Comment:
Please give me some directions on how I can improve the naming scheme.

Do you mean to use the name `malloc` instead of `allocate`? I could do that, but I would have to use the `-fno-builtin` flag, because right now gcc complains:
> include/memory_allocator.h:21:7: error: conflicting types for built-in function ‘malloc’ [-Werror]